src/hooks/usePing.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import { ping } from '../utils';
+interface IntranetCheck {
+    isSourceAvailable: boolean;
+}
+
+interface CachedPing {
+    pingTask: Promise<boolean>;
+    timestamp: Date;
+}
+
+const cachedPings: Record<string, CachedPing> = {};
+
+function dateDifferenceInSeconds(date1?: Date, date2?: Date): number {
```

Comment:
should be moved to formatTimeHelpers.ts and tested

Reply:
Good point, seems generic enough!

Reply:
Moved and tested.